pvsreact/src/pages/Signup.js

```diff
@@ -0,0 +1,55 @@
+import dog from "../images/Sign-In-1.jpg"
+import { FaHome } from "react-icons/fa";
+import styles from '../CSS/style.module.css'
+import { useNavigate } from "react-router-dom";
+import { useState } from "react";
+import { FaGoogle } from "react-icons/fa";
+import { Link } from "react-router-dom";
+
+
+export default function Signup() {
+
+    const navigate = useNavigate();
+    const [userName, setUserName] = useState("");
+    const [password, setPassword] = useState("");
+    const [confirmedPassword, setConfirmedPassword] = useState("");
+    const [email, setEmail] = useState("");
+    const handleUserName = (e) => {
+        setUserName(e.target.value);
+    }
+    const handlePassword = (e) => {
+        setPassword(e.target.value);
+    }
+    const handleConfirmedPassword = (e) => {
+        setConfirmedPassword(e.target.value);
+    }
+    const handleEmail = (e) => {
+        setEmail(e.target.value);
+    }
+    const handleSubmit = (e) => {
+        e.preventDefault();
+        console.log(userName);
+        console.log(password);
+        console.log(email);
+        navigate('/SignupController', { replace: true, state: { userName, password, email } });
+    }
+    return (
+        <>
+            <div className={styles.signin}>
+                <div className={styles.content}>
+                    <form onSubmit={handleSubmit}>
+                        <Link to="/Signin" className={styles.home}><FaHome></FaHome> Return</Link>
+                        <div className={styles.headup}>Sign up</div>
+                        <input type="text" placeholder="Username" value={userName} onChange={handleUserName} required></input>
```

Comment:
if the backend doesn't check for the email format validity , there should be a regex to handle that

Reply:
The email input handle the regex